src/crimson/os/seastore/onode_manager/staged-fltree/node.cc

```diff
@@ -17,6 +17,14 @@
 
 SET_SUBSYS(seastore_onode);
 
+namespace fmt {
+
+template <typename T> auto ptr(const ::boost::intrusive_ptr<T>& p) -> const void* {
```

Comment:
Is there a benefit from `auto ... -> const void*`?

Reply:
what exactly are you referring? the trailing return type? i am following https://github.com/fmtlib/fmt/blob/7df30f91aee5444a733cec0b911d21cebdeb62ae/include/fmt/format.h#L3887-L3897 . if this is not encouraged, i will use the old-style `const void* ptr(...)`. in this case, no benefit in particular. but like the C++ Code Guidelines explains in [ `{}` ](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list). this form is more expressive, so might be advisable to just use it everywhere.

Reply:
guess that's why clang-tidy has this check, see https://clang.llvm.org/extra/clang-tidy/checks/modernize/use-trailing-return-type.html

Reply:
changed it to the old style. yeah, it increases the mental load of human readers who are not used to the trailing return type. but i think it's all about the custom of how people read/parse/understand/the code .  one can also read it like "a function named foo which takes x and y and returns bar"